core/field_textdropdown.js

```diff
@@ -0,0 +1,149 @@
+ *     in the text field.
+ * @extends {Blockly.FieldTextInput}
+ * @constructor
+ */
+Blockly.FieldTextDropdown = function(text, menuGenerator, opt_validator, opt_restrictor) {
+  this.menuGenerator_ = menuGenerator;
+  Blockly.FieldDropdown.prototype.trimOptions_.call(this);
+  Blockly.FieldTextDropdown.superClass_.constructor.call(this, text, opt_validator, opt_restrictor);
+  this.addArgType('textdropdown');
+};
+goog.inherits(Blockly.FieldTextDropdown, Blockly.FieldTextInput);
+
+/**
+ * Install this text drop-down field on a block.
+ */
+Blockly.FieldTextDropdown.prototype.init = function() {
+  Blockly.FieldTextDropdown.superClass_.init.call(this);
+  // Add dropdown arrow: "option ▾" (LTR) or "▾ אופציה" (RTL)
+  // Positioned on render, after text size is calculated.
+  if (!this.arrow_) {
+    /** @type {Number} */
+    this.arrowSize_ = 12;
+    /** @type {Number} */
+    this.arrowX_ = 0;
+    /** @type {Number} */
+    this.arrowY_ = 11;
+    this.arrow_ = Blockly.createSvgElement('image', {
+      'height': this.arrowSize_ + 'px',
+      'width': this.arrowSize_ + 'px'
+    });
+    this.arrow_.setAttributeNS('http://www.w3.org/1999/xlink',
+        'xlink:href', Blockly.mainWorkspace.options.pathToMedia + 'dropdown-arrow-dark.svg');
+    this.arrow_.style.cursor = 'pointer';
+    this.fieldGroup_.appendChild(this.arrow_);
+    this.mouseUpWrapper_ =
+        Blockly.bindEvent_(this.arrow_, 'mouseup', this,
+            this.showDropdown_);
+  }
+  // Prevent the drop-down handler from changing the field colour on open.
+  this.disableColourChange_ = true;
+};
+
+/**
+ * Close the input widget if this input is being deleted.
+ */
+Blockly.FieldTextDropdown.prototype.dispose = function() {
+  if (this.mouseUpWrapper_) {
+    Blockly.unbindEvent_(this.mouseUpWrapper_);
+    this.mouseUpWrapper_ = null;
+  }
+  Blockly.FieldTextDropdown.superClass_.dispose.call(this);
+};
+
+/**
+ * Return a list of the options for this dropdown.
+ * See: Blockly.FieldDropDown.prototype.getOptions_.
+ * @return {!Array.<!Array.<string>>} Array of option tuples:
+ *     (human-readable text, language-neutral name).
+ * @private
+ */
+Blockly.FieldTextDropdown.prototype.getOptions_ = function() {
+  return Blockly.FieldDropdown.prototype.getOptions_.call(this);
+};
+
+/**
+ * Position a drop-down arrow at the appropriate location at render-time.
+ * See: Blockly.FieldDropDown.prototype.positionArrow.
+ * @param {number} x X position the arrow is being rendered at, in px.
+ * @return {number} Amount of space the arrow is taking up, in px.
+ */
+Blockly.FieldTextDropdown.prototype.positionArrow = function(x) {
+  return Blockly.FieldDropdown.prototype.positionArrow.call(this, x);
+};
+
+/**
+ * If the drop-down isn't open, show the text editor.
+ */
+Blockly.FieldTextDropdown.prototype.showEditor_ = function() {
+  if (!this.dropDownOpen_) {
+    Blockly.FieldTextDropdown.superClass_.showEditor_.call(this, null, null,
```

Comment:
the superclass of the text drop-down is FieldTextInput - it's showing the text editor.